src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs

```diff
@@ -859,35 +859,79 @@ public async Task SslStream_UntrustedCaWithCustomCallback_Throws(bool customCall
 
-            using (X509Store store = new X509Store(StoreName.CertificateAuthority, StoreLocation.CurrentUser))
+            using (X509Store store = new X509Store(storeName, StoreLocation.CurrentUser))
             {
                 // add chain certificate so we can construct chain since there is no way how to pass intermediates directly.
                 store.Open(OpenFlags.ReadWrite);
                 store.AddRange(clientChain);
                 store.Close();
             }
 
-            using (var chain = new X509Chain())
+            _output.WriteLine("Certificates added to {0}", storeName);
+
+            // make sure we can build chain. There may be some race conditions after certs being added to the store.
+            int retries = 10;
+            int delay = 100;
+            while (retries > 0)
             {
-                chain.ChainPolicy.VerificationFlags = X509VerificationFlags.AllFlags;
-                chain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
-                chain.ChainPolicy.DisableCertificateDownloads = false;
-                bool chainStatus = chain.Build(clientCertificate);
-                // Verify we can construct full chain
-                if (chain.ChainElements.Count < clientChain.Count)
+                using (var chain = new X509Chain())
                 {
-                    throw new SkipTestException($"chain cannot be built {chain.ChainElements.Count}");
+                    chain.ChainPolicy.VerificationFlags = X509VerificationFlags.AllFlags;
+                    chain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
+                    chain.ChainPolicy.DisableCertificateDownloads = true;
+                    bool chainStatus = chain.Build(clientCertificate);
+                    _output.WriteLine("round {0}: chain.Build: finished as {1} with {2} certificates {3}", retries, chainStatus, chain.ChainElements.Count, DateTime.Now);
+                    if (chainStatus && chain.ChainElements.Count >= clientChain.Count)
+                    {
+                        break;
+                    }
                 }
+
+                Thread.Sleep(delay);
+                delay *= 2;
+                retries--;
             }
 
+            if (retries == 0)
+            {
+                using (var chain = new X509Chain())
+                {
+                    chain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
+                    chain.Build(clientCertificate);
+                    foreach (X509ChainElement element in chain.ChainElements)
+                    {
+                        _output.WriteLine($"{element.Certificate.Subject} length {element.ChainElementStatus.Length}");
+                        foreach (X509ChainStatus status in element.ChainElementStatus)
+                        {
+                            _output.WriteLine($"  Status:  {status.Status}: {status.StatusInformation}");
+                        }
+                    }
```

Comment:
Logging items from chain.ChainStatus may also be useful, in case there's an overall code it presented that didn't map to any individual element

Reply:
```
        CN=SslStream_ClientCertificate_SendsChain, O=SslStream_ClientCertificate_SendsChain length 1
          Status:  PartialChain: One or more certificates required to validate this certificate cannot be found.
          ChainStatus: PartialChain One or more certificates required to validate this certificate cannot be found..
```

it seems like it cannot find the part certs.
That are in store dump AFAIK
```
        CN=A Revocation Test CA 0, O=SslStream_ClientCertificate_SendsChain
        CN=A Revocation Test Root, O=SslStream_ClientCertificate_SendsChain
```

Reply:
Huh.  I certainly would have expected the default keychain to be involved in a chain build; but I guess not.  We probably don't have tests like this in the X509 suite because we don't like messing with system state.

Since there's not a flow where the client can provide their own chain context, you might just need to have this test leave the responder running, so the intermediate can be fetched from AIA.  (Overload GenerateCertificates to out the responder instead of disposing it)

Reply:
I did many runs on my local machines and it works fine. But there is something unusual about the CI machine ;(
I was thinking about making the test optional again but we lost coverage before and it hid valid issue. 

And I can try the AIA. I did not like it for the servers as it make some of the test less predicable but I can try it for client.

Reply:
> I did many runs on my local machines and it works fine. But there is something unusual about the CI machine ;(

Now _that_ sounds familiar :).  IIRC it doesn't have a UI session, just an SSH session, and that makes some of the keychain systems behave oddly.